### PR TITLE
fixes #25597 - clear checksum task works with root repo change

### PR DESCRIPTION
**Pull request by stbenjam**
I think this preserves the meaning the change, in light of the conversion to root repository but please look closely, I'm not entirely all that familiar with pulp's internals.

### CODE REVIEW

File: lib/katello/tasks/upgrades/3.8/clear_checksum_type.rake [hunk 1]
```diff
@@ -5,24 +5,22 @@ namespace :katello do
...
+                    if repo.find_distributor[:config]&.delete(:checksum_type)
+                      Katello.pulp_server.resources.repository.update_distributor(
+                        repo.pulp_id, repo.find_distributor[:id], repo.find_distributor[:config])
+                    end
+                  rescue RestClient::ResourceNotFound
+                  end
                 end
               end
-
-              if repo.library_instance?
```
[the remark this answers is not in the document]
Comment (jturel) [reply lib/katello/tasks/upgrades/3.8/clear_checksum_type.rake]: Ah, that field is gone as well!

File: lib/katello/tasks/upgrades/3.8/clear_checksum_type.rake [hunk 2]
```diff
@@ -5,24 +5,22 @@ namespace :katello do
...
+              if root_repo.on_demand? && root_repo.url.present?
+                root_repo.update_attribute(:checksum_type, nil)
 
-                if repo.find_distributor[:config]&.delete(:checksum_type)
-                  Katello.pulp_server.resources.repository.update_distributor(
-                    repo.pulp_id, repo.find_distributor[:id], repo.find_distributor[:config])
+                root_repo.repositories.each do |repo|
+                  begin
+                    if repo.find_distributor[:config]&.delete(:checksum_type)
+                      Katello.pulp_server.resources.repository.update_distributor(
```
Comment (akofink) [inline lib/katello/tasks/upgrades/3.8/clear_checksum_type.rake]: The "new way" to access this is `SmartProxy.pulp_master.pulp_api.resources.repository`

File: lib/katello/tasks/upgrades/3.8/clear_checksum_type.rake [hunk 3]
```diff
@@ -5,24 +5,22 @@ namespace :katello do
...
 
-        Katello::Repository.yum_type.find_each do |repo|
-          repo.transaction do
+        Katello::RootRepository.yum_type.find_each do |root_repo|
+          root_repo.transaction do
             begin
-              if repo.on_demand? && repo.url.present?
-                repo.update_attribute(:checksum_type, nil)
+              if root_repo.on_demand? && root_repo.url.present?
+                root_repo.update_attribute(:checksum_type, nil)
```
Comment (akofink) [inline lib/katello/tasks/upgrades/3.8/clear_checksum_type.rake]: I think we also need to do this for each `repo.saved_checksum_type`?

Comment (akofink) [reply lib/katello/tasks/upgrades/3.8/clear_checksum_type.rake]: I'm also not sure if we need to clear out the root repo's checksum type. @jlsherrill would know

Comment (jlsherrill) [reply lib/katello/tasks/upgrades/3.8/clear_checksum_type.rake]: /me digs around in his memory.  Yes we need to set the saved_checksum_type to nil on the repositories.  Do we need to update the root_repo's checksum type....  I believe so.  

Good catch @akofink !

Comment (stbenjam) [reply lib/katello/tasks/upgrades/3.8/clear_checksum_type.rake]: Thanks, handled below when going through the individual repos

**Comment by jturel** [comment]
trailing whitespace :(

**Comment by stbenjam** [comment]
:man_facepalming:

**Comment by stbenjam** [comment]
[test katello]

**Comment by stbenjam** [comment]
[test katello]

**Pull request merged**